operators/src/processing/meteosat/radiance.rs

```diff
@@ -191,47 +232,19 @@ where
         ctx: &'a dyn QueryContext,
     ) -> Result<BoxStream<'a, Result<Self::Output>>> {
         let src = self.source.query(query, ctx).await?;
-        Ok(src
-            .map(move |tile| match tile {
-                Ok(tile) if tile.grid_array.is_empty() => Ok(RasterTile2D::new_with_properties(
-                    tile.time,
-                    tile.tile_position,
-                    tile.global_geo_transform,
-                    EmptyGrid::new(tile.grid_array.grid_shape(), self.no_data_value).into(),
-                    tile.properties,
-                )),
-                Ok(tile) => {
-                    let mg = tile.grid_array.into_materialized_grid();
-
-                    let mut out = Grid2D::new(
-                        mg.grid_shape(),
-                        vec![self.no_data_value; mg.data.len()],
-                        Some(self.no_data_value),
-                    )
-                    .expect("raster creation must succeed");
-
-                    let offset = tile.properties.number_property::<f32>(&self.offset_key)?;
-                    let slope = tile.properties.number_property::<f32>(&self.slope_key)?;
-                    let tgt = &mut out.data;
-
-                    for (idx, v) in mg.data.iter().enumerate() {
-                        if !mg.is_no_data(*v) {
-                            let val: PixelOut = (*v).as_();
-                            tgt[idx] = offset + val * slope;
-                        }
-                    }
-
-                    Ok(RasterTile2D::new_with_properties(
-                        tile.time,
-                        tile.tile_position,
-                        tile.global_geo_transform,
-                        out.into(),
-                        tile.properties,
-                    ))
-                }
-                Err(e) => Err(e),
+        let rs = src
+            .map(move |tile| {
+                tile.and_then(|t| {
+                    let offset = t.properties.number_property::<f32>(&self.offset_key)?;
```

Comment:
could these be determined outside of the `map`?

Reply:
no you need the tile for that